errs/error.go

```diff
@@ -268,15 +270,11 @@ func NewError(status int, err error, format string, args ...interface{}) error {
 func NewErr(status int, err error, opts ...Option) error {
 	var e *Error
 	if !errors.As(err, &e) {
-		if sc, ok := err.(render.StatusCodedError); ok {
-			e = &Error{Status: sc.StatusCode(), Err: err}
+		var ste render.StatusCodedError
+		if errors.As(err, &ste) {
+			e = &Error{Status: ste.StatusCode(), Err: err}
 		} else {
-			cause := errors.Cause(err)
-			if sc, ok := cause.(render.StatusCodedError); ok {
-				e = &Error{Status: sc.StatusCode(), Err: err}
-			} else {
-				e = &Error{Status: status, Err: err}
-			}
```

Comment:
Kinda curious if there are cases that are now not handled like a `StatusCodedError` after the change. We may not have been testing those, or they just don't exist. Getting rid of the `.Cause` has my vote, though.